5-network/01-fetch/article.md

```diff
@@ -1,158 +1,158 @@
-- **`response.formData()`** -- return the response as `FormData` object (explained in the [next chapter](info:formdata)),
-- **`response.blob()`** -- return the response as [Blob](info:blob) (binary data with type),
-- **`response.arrayBuffer()`** -- return the response as [ArrayBuffer](info:arraybuffer-binary-arrays) (low-level representaion of binary data),
-- additionally, `response.body` is a [ReadableStream](https://streams.spec.whatwg.org/#rs-class) object, it allows to read the body chunk-by-chunk, we'll see an example later.
+- **`response.text()`** -- odczytaj odpowiedź i zwróć jako tekst,
+- **`response.json()`** -- odczytaj odpowiedź i zwróć jako JSON,
+- **`response.formData()`** -- zwróć odpowiedź jako obiekt typu `FormData` (wyjaśnienie w [następnym rozdziale](info:formdata)),
+- **`response.blob()`** -- zwróć odpowiedź jako [Blob](info:blob) (dane binarne z typem),
+- **`response.arrayBuffer()`** -- zwróć odpowiedź jako [ArrayBuffer](info:arraybuffer-binary-arrays) (niskopoziomowa reprezentacja danych binarnych),
+- ponadto `response.body` jest sam w sobie obiektem typu [ReadableStream](https://streams.spec.whatwg.org/#rs-class), co pozwala na odczytywanie go kawałek po kawałku, co zostanie pokazane w kolejnym przykładzie.
 
-For instance, let's get a JSON-object with latest commits from GitHub:
+Pobierzmy dla przykładu obiekt JSON z ostatnimi commitami z GitHuba.
 
 ```js run async
 let url = 'https://api.github.com/repos/javascript-tutorial/en.javascript.info/commits';
 let response = await fetch(url);
 
 *!*
-let commits = await response.json(); // read response body and parse as JSON
+let commits = await response.json(); // odczytaj ciało odpowiedzi i zwróć jako JSON
 */!*
 
 alert(commits[0].author.login);
 ```
 
-Or, the same without `await`, using pure promises syntax:
+Bądź to samo, ale bez `await`, a jedynie za pomocą czystej składni obietnic:
 
 ```js run
 fetch('https://api.github.com/repos/javascript-tutorial/en.javascript.info/commits')
   .then(response => response.json())
   .then(commits => alert(commits[0].author.login));
 ```
 
-To get the reponse text, `await response.text()` instead of `.json()`:
+Aby pobrać odpowiedź jako tekst, użyj `await response.text()` zamiast `.json()`:
 
 ```js run async
 let response = await fetch('https://api.github.com/repos/javascript-tutorial/en.javascript.info/commits');
 
-let text = await response.text(); // read response body as text
+let text = await response.text(); // odczytaj ciało odpowiedzi jako tekst
 
 alert(text.slice(0, 80) + '...');
 ```
 
-As a show-case for reading in binary format, let's fetch and show a logo image of ["fetch" specification](https://fetch.spec.whatwg.org) (see chapter [Blob](info:blob) for details about operations on `Blob`):
+Aby zaprezentować odczyt danych w formacie binarnym, pobierzmy obraz logo [specyfikacji "fetch"](https://fetch.spec.whatwg.org) (patrz rozdział [Blob](info:blob) odnośnie operacji na obiekcie typu `Blob`):
```

Comment:
```suggestion
Aby zaprezentować odczyt danych w formacie binarnym, pobierzmy obraz logo [specyfikacji "fetch"](https://fetch.spec.whatwg.org) (patrz rozdział [Blob](info:blob) odnośnie operacji na obiekcie klasy `Blob`):
```

Reply:
No właśnie kurcze, ponieważ autor nie napisał z Blob(), w sensie konstruktora, pomyślałem raczej o obiekcie niż klasie (na marginesie w dokumentacji jest zasadniczo interfejs).

Reply:
Ok, ale chodzi mi o to, że w JS-ie typów jest tylko kilka: string, number, boolean, object, function itp. `Blob` nie jest typem *per se*, tylko pewnym konkretnym rodzajem obiektu powstałego z konstruktora `Blob()` (tu masz rację). Zwykle w dokumentacjach i książkach z programowania obiektowego mówi się, że jest "klasa Blob" albo "interfejs Blob". JS nie jest stricte obiektowy, ale... to na pewno nie będzie "typ" :D